src/iarc7_motion/iarc_tasks/position_holder.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+
+# A helper class for a task that will spit out a velocity that's appropriate
+# to reach and hold a desired position
+
+import math
+import threading
+
+import rospy
+
+from geometry_msgs.msg import TwistStamped
+from nav_msgs.msg import Odometry
+
+class PositionHolder():
+    def __init__(self, x=None, y=None):
+        update_rate = rospy.get_param('~update_rate', False)
+        #self._update_period = 1.0/update_rate
+        self._update_period = 0.1
+        self._lock = threading.RLock()
+        self._odometry = None
+        self._hold_x = x
+        self._hold_y = y
+        self._max_acceleration = rospy.get_param('~max_translation_acceleration', 0.0)
+        self._max_speed = rospy.get_param('~max_translation_speed', 0.0)
+        self._position_tolerance = rospy.get_param('~translation_position_hold_tolerance', 0.0)
+        self._current_velocity_sub = rospy.Subscriber('/odometry/filtered',
+                                              Odometry,
+                                              self._current_velocity_callback)
+
+    def get_xy_hold_response(self, current_x, current_y, z_velocity=None):
+        with self._lock:
+            if self._odometry is not None:
+                delta_x = current_x - self._hold_x
+                delta_y = current_y - self._hold_y
+                response = self._get_next_acceleration_target_trapezoidal(delta_x,
+                                                                          delta_y,
+                                                                          self._odometry.twist,
+                                                                          z_velocity=z_velocity)
+            else:
+                rospy.logerr('get_xy_hold_response called before odometry published')
+
+            return response
+
+    def _current_velocity_callback(self, odometry):
+        with self._lock:
+            if odometry.header.frame_id == 'map' and odometry.child_frame_id == 'level_quad':
```

Comment:
There shouldn't be odometry for different robots on the same topic (for example each roomba has its odometry published by the simulator on /sim/roomba#/odom).  So in theory yes, we could have multiple robot_localization nodes publishing odometry for different robots, but they should never be on that topic